db/document.go

```diff
@@ -411,15 +411,15 @@ func unmarshalDocumentWithXattr(ctx context.Context, docid string, data []byte,
 	return doc, nil
 }
 
-// Unmarshals just a document's sync metadata from JSON data.
+// Unmarshals just a document's sync metadata from JSON data. Errors if there is no _sync attribute in the data.
 // (This is somewhat faster, if all you need is the sync data without the doc body.)
 func UnmarshalDocumentSyncData(data []byte, needHistory bool) (*SyncData, error) {
 	var root documentRoot
 	if needHistory {
 		root.SyncData = &SyncData{History: make(RevTree)}
 	}
 	if err := base.JSONUnmarshal(data, &root); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%w: %w", base.ErrNoInlineSyncData, err)
```

Comment:
The error name here (and the comment describing the function) doesn't look correct to me.  It's only going to return error here if `data` is not a JSON object.   For cases where it's a JSON object but doesn't include _sync, or even if _sync is present and it's not SyncData, I don't think we'll return error here.

Reply:
We don't have those cases in the unit test, might want to add those and ensure they are being handled (and logged) as expected.

Reply:
I think this is still correct behavior demonstrated in the test, but let me know if this is otherwise. Since this unmarshals into `documentRoot`, you'll see a marshal error in the case of no `_sync` attribute.

The case that is ambiguously worried is where `_sync` is present but is not `SyncData` it will return `ErrNoInlineSyncData`, which is sort of misleading.

I added additional cases to at least regression test what happens if we have `_sync` xattr and also `_sync` inline (we ignore inline).